### PR TITLE
Cristin series

### PR DESCRIPTION
**Pull request by axthosarouris**
Persist date for Cristin series and Journals

### CODE REVIEW

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/constants/MappingConstants.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
     public static final String NVA_API_DOMAIN = "https://api." + readDomainName();
     public static final String NSD_PROXY_PATH = "publication-channels";
     public static final String NSD_PROXY_PATH_JOURNAL = "journal";
+    public static final String NSD_PROXY_PATH_PUBLISHER = "publisher";
```
Comment (axthosarouris) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/constants/MappingConstants.java:23]: will be used in next PR

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java [hunk 2]
```diff
@@ -188,15 +182,6 @@ private PublicationContext buildPublicationContext()
         return null;
     }
 
-    private Book buildBookForPublicationContext() throws InvalidIsbnException {
```
Comment (axthosarouris) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java:191]: moved to NvaBookBuilder

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMappingModule.java [hunk 3]
```diff
@@ -1,22 +0,0 @@
-package no.unit.nva.cristin.mapper;
```
Comment (axthosarouris) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMappingModule.java:1]: moved to another package

File: buildSrc/gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.0.2-all.zip
```
Comment (brinxmat) [inline buildSrc/gradle/wrapper/gradle-wrapper.properties:3]: Any reason not to bump all the way?

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/PeriodicalBuilder.java [hunk 5]
```diff
@@ -34,11 +32,9 @@ private Periodical createUnconfirmedJournal() {
     }
 
     private Periodical createJournal() {
-        URI journalUri = new UriWrapper(NVA_API_DOMAIN).addChild(MappingConstants.NSD_PROXY_PATH)
-            .addChild(MappingConstants.NSD_PROXY_PATH_JOURNAL)
-            .addChild(cristinObject.getJournalPublication().getJournal().getNsdCode().toString())
-            .addChild(cristinObject.getPublicationYear().toString())
-            .getUri();
+        Integer nsdCode = cristinObject.getJournalPublication().getJournal().getNsdCode();
+        int publicationYear = cristinObject.getPublicationYear();
```
Comment (brinxmat) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/PeriodicalBuilder.java:36]: Maybe just me, but is this the correct target — from the conversation with Kari and Daniel, I'd assumed this should be "reporting year", but I may have misunderstood

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/nva/CristinMappingModule.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+    protected PublishingHouse buildUnconfirmedPublisher() {
+        return new UnconfirmedPublisher(extractPublisherName());
+    }
+
+
+    private String extractPublisherName() {
+        return extractCristinBookReport().getPublisherName();
+    }
+
```
Comment (brinxmat) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/nva/CristinMappingModule.java:46]: Remove surplus space

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/nva/NvaBookBuilder.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    private String constructSeriesNumber() {
+        String volume = Optional.of(cristinObject)
+            .map(CristinObject::getBookOrReportMetadata)
+            .map(CristinBookOrReportMetadata::getVolume)
+            .orElse(null);
+        String issue = Optional.of(cristinObject)
+            .map(CristinObject::getBookOrReportMetadata)
+            .map(CristinBookOrReportMetadata::getIssue)
+            .orElse(null);
+        return String.format("Volume:%s;Issue:%s", volume, issue);
```
Comment (brinxmat) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/nva/NvaBookBuilder.java:34]: This is the concatenation we were talking about on Thursday 16th Nov 2021, right?

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/nva/NvaBookSeriesBuilder.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+import no.unit.nva.cristin.mapper.Nsd;
+import no.unit.nva.model.contexttypes.BookSeries;
+import no.unit.nva.model.contexttypes.Series;
+import no.unit.nva.model.contexttypes.UnconfirmedSeries;
+
+public class NvaBookSeriesBuilder extends CristinMappingModule {
+
+    public NvaBookSeriesBuilder(CristinObject cristinObject) {
+        super(cristinObject);
+
```
Comment (brinxmat) [inline cristin-import/src/main/java/no/unit/nva/cristin/mapper/nva/NvaBookSeriesBuilder.java:19]: Remove space

File: cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java [hunk 9]
```diff
@@ -276,14 +241,82 @@ public static String randomIsbn13() {
...
+            .withVolume(randomString())
+            .build();
+    }
+
     private static String random7DigitNumber() {
-        return String.valueOf((int)(Math.random() * 9000000) + 1000000);
+        return String.valueOf((int) (Math.random() * 9000000) + 1000000);
+    }
+
+    private static String createValidIssn(String issnMissingChecksum) {
```
Comment (brinxmat) [inline cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java:266]: Have I seen this code somewhere before in a shared lib?

File: cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java [hunk 10]
```diff
@@ -432,43 +465,45 @@ private static CristinObject createObjectWithCristinContributorRoleCode(Integer
...
     private static CristinJournalPublication randomJournalPublictaion() {
         return CristinJournalPublication.builder()
-                .withJournal(randomCristinJournalPublicationJournal())
-                .withPagesBegin("1")
-                .withPagesEnd(String.valueOf(smallRandomNumber()))
-                .withVolume(String.valueOf(smallRandomNumber()))
-                .withDoi(randomDoiString())
-                .build();
+            .withJournal(randomCristinJournalPublicationJournal())
+            .withPagesBegin("1")
```
Comment (brinxmat) [inline cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java:481]: May as well extract this magic so it is done

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/nva/NvaBookBuilder.java (hunk 7 above)
Comment (axthosarouris) [reply cristin-import/src/main/java/no/unit/nva/cristin/mapper/nva/NvaBookBuilder.java:34]: yes

File: cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java (hunk 9 above)
Comment (axthosarouris) [reply cristin-import/src/test/java/no/unit/nva/cristin/CristinDataGenerator.java:266]: in nva-publication-api. Maybe we should create a small class in testutils for creatting random data

File: cristin-import/src/main/java/no/unit/nva/cristin/mapper/PeriodicalBuilder.java (hunk 5 above)
Comment (axthosarouris) [reply cristin-import/src/main/java/no/unit/nva/cristin/mapper/PeriodicalBuilder.java:36]: reporting year is not always present. I can try to fix this.

File: buildSrc/gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (axthosarouris) [reply buildSrc/gradle/wrapper/gradle-wrapper.properties:3]: This was an accident. I will upgrade in separate PR

**Review by brinxmat — APPROVED** [review]

**Pull request merged**